test/collections_test.rb

```diff
@@ -0,0 +1,119 @@
+require_relative "./collections_test_helper"
+
+describe "collections" do
+  collections.each do |collection|
+    describe "#{collection} collection" do
+      it "has a valid name" do
+        assert valid_collection?(collection), invalid_collection_message(collection)
+      end
+
+      it "does not include emoji outside of description" do
+        metadata = metadata_for(collections_dir, collection) || {}
+
+        fields = %w[created_by display_name collection]
+        fields.each do |field|
+          if value = metadata[field].to_s
+            assert value == value.gsub(EMOJI_REGEX, ""),
+                   "#{field} should not include emoji:\n\t#{value}"
+          end
+        end
+      end
+
+      it "has valid items" do
+        metadata = metadata_for(collections_dir, collection) || {}
+        items = metadata["items"]
+        invalid_slugs = []
+        items.each do |item|
+          begin
+            URI.parse(item)
+          rescue URI::InvalidURIError
+            invalid_slugs << item
+          end
+        end
+        assert_empty invalid_slugs, "Invalid item slugs #{invalid_slugs}"
+      end
+
+      it "has an index.md" do
+        path = File.join(collections_dir, collection, "index.md")
+
+        assert File.file?(path), "expected #{path} to be a file"
+      end
+
+      it "has no unexpected files or directories" do
+        files = Dir["#{collections_dir}/#{collection}/**/*"].reject do |entry|
+          file_name = File.basename(entry)
+          image_files = possible_image_file_names_for_collection(collection)
```

Comment:
This can be pulled outside of the `reject` loop since it only depends on `collection`.